# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/DeleteRangeTransaction.cpp
# Commit: ce2087ddab5c
# Full Hash: ce2087ddab5c2efcbce3b52da02b3363a37d6813
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-04-04 09:29:13
# Regressor Bug: 1619914
# File Overlap Count: 1
# Description:
#   Bug 1619914 - part 1: Make each transaction class grab their members with local variable before touching the DOM tree r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D69302
# ==============================================================================

diff -r 1c8115a9a684 -r ce2087ddab5c editor/libeditor/DeleteRangeTransaction.cpp
--- a/editor/libeditor/DeleteRangeTransaction.cpp	Fri Apr 03 10:21:33 2020 +0000
+++ b/editor/libeditor/DeleteRangeTransaction.cpp	Fri Apr 03 08:30:37 2020 +0000
@@ -130,7 +130,7 @@
   }
 
   // see what kind of node we have
-  if (RefPtr<Text> textNode = Text::FromNode(aStart.Container())) {
+  if (Text* textNode = Text::FromNode(aStart.Container())) {
     // if the node is a chardata node, then delete chardata content
     int32_t numToDel;
     if (aStart == aEnd) {
@@ -193,7 +193,7 @@
     return NS_ERROR_NOT_AVAILABLE;
   }
 
-  RefPtr<Text> textNode = Text::FromNode(aPoint.Container());
+  Text* textNode = Text::FromNode(aPoint.Container());
   if (!textNode) {
     return NS_OK;
   }
@@ -243,13 +243,13 @@
   }
 
   for (; !subtreeIter.IsDone(); subtreeIter.Next()) {
-    nsCOMPtr<nsINode> node = subtreeIter.GetCurrentNode();
-    if (NS_WARN_IF(!node)) {
-      return NS_ERROR_NULL_POINTER;
+    nsINode* node = subtreeIter.GetCurrentNode();
+    if (NS_WARN_IF(!node) || NS_WARN_IF(!node->IsContent())) {
+      return NS_ERROR_FAILURE;
     }
 
     RefPtr<DeleteNodeTransaction> deleteNodeTransaction =
-        DeleteNodeTransaction::MaybeCreate(*mEditorBase, *node);
+        DeleteNodeTransaction::MaybeCreate(*mEditorBase, *node->AsContent());
     // XXX This is odd handling.  Even if some nodes in the range are not
     //     editable, editor should append transactions because they could
     //     at undoing/redoing.  Additionally, if the transaction needs to